src/scripts/js-telephone-input.js

```diff
@@ -0,0 +1,487 @@
+        this.preventKeyup = false;
+        this.phoneNumber = null;
+        this.exampleNumber = null;
+
+        //Validate User Default Value
+        validateInitialValue(this);
+
+        //Define field events and attributes
+        addEvent(this.field, 'keyup', format, this);
+        addEvent(this.field, 'click', format, this);
+        addEvent(this.field, 'keypress', checkPressedKey, this);
+        addEvent(this.field, 'blur', this.setInputsValue, this);
+        setExampleNumber(this.parameters.country, this);
+    };
+
+    //----------------------------------------------------
+    // Public Methods
+    //----------------------------------------------------
+
+    // Sets the value on the hidden inputs
+    jsTelephoneInput.prototype.setInputsValue = function(event, component) {
+        if(this instanceof global.component) {
+            component = this;
+        }
+        var valid = component.isValid(),
+            telephone = component.getTelephone(),
+            numberInput = document.getElementById(component.parameters.number),
+            areaCodeInput = document.getElementById(component.parameters.areaCode),
+            fullNumberInput = document.getElementById(component.parameters.fullNumber);
+
+        if(numberInput) {
+            numberInput.value = (valid) ? telephone.telephone : '';
+        }
+        if(areaCodeInput) {
+            areaCodeInput.value = (valid) ? telephone.areaCode: '';
+        }
+        if(fullNumberInput) {
+            fullNumberInput.value = (valid) ? telephone.fullTelephone: '';
+        }
+
+        if (valid) {
+            component.validations.hide(component.validations.info.zero);
+            return true;
+        }
+        return false;
+    };
+
+    // Gets the telephone areaCode and number in json format
+    jsTelephoneInput.prototype.getTelephone = function() {
+        var areaCodeLimit = this.field.value.indexOf(' ');
+        //If the telephone does not contain spaces, cut the areaCode using the "-"
+        if(areaCodeLimit === -1) {
+            areaCodeLimit = this.field.value.indexOf('-');
+        }
+        else {
+            //If there are multiple spaces, use the first two parts as the areaCode.
+            if((this.field.value.match(/\s/g) || []).length > 2) {
+                areaCodeLimit = this.field.value.indexOf(' ', areaCodeLimit + 1);
+            }
+        }
+        return {
+            country: this.parameters.country,
+            countryCode: this.LibPhoneNumber.getCountryCodeForRegionCode(this.parameters.country),
+            fullTelephone: stripValue(this.field.value).join(''),
+            areaCode: stripValue(this.field.value.substring(0, areaCodeLimit)).join(''),
+            telephone: stripValue(this.field.value.substring(areaCodeLimit + 1, this.field.value.length)).join('')
+        }
+    };
+
+    // Checks if the number is valid
+    jsTelephoneInput.prototype.isValid = function() {
+        if (this.field.value.length === 0) {
```

Comment:
@joelalejandro Acá el comportamiento es distinto.
Si el campo es no requerido, pueden pasar dos cosas.
Si el usuario no escribió nada, el resultado es válido.
Si el usuario escribió algo, puede ser true o false, dependiendo si escribió algo válido o no. Entonces podríamos hacer algo así:
`if(!this.parameters.required && this.field.value.length === 0) {
            return true;
        }
        return this.phoneNumber.isValid();`